doc/jobs/ci_jobs.rst

```diff
@@ -0,0 +1,188 @@
+*CI* jobs
+=========
+
+A *CI* job is used for continuous integration in batch.
+It builds the code and runs the tests to check for regressions.
+It operates on many source repositories as specified in one or more ``.repos``
+files, and is triggered to produce a nightly comprehensive archive, and can
+also be triggered manually to quickly test changes on a select set of
+packages.
+
+Each build is performed within a clean environment (provided by a Docker
+container) which only contains the specific dependencies of packages in the
+repository as well as tools needed to perform the build and any packages
+specified as part of the ``foundation_packages``.
+
+Entry points
+------------
+
+The following scripts are the entry points for *CI* jobs.
+The scripts operate on the default ``.repos`` file in the ROS build farm
+configuration, unless specified otherwise:
+
+* **generate_ci_maintenance_jobs.py** generates a set of jobs on the farm
+  which will perform maintenance tasks.
+
+  * The ``reconfigure-jobs`` job will (re-)configure the *CI* nightly and
+    overlay jobs on a regular basis (e.g. once every day).
+
+* **generate_ci_jobs.py** invokes *generate_ci_job.py* for both the nightly
+  configuration as well as the overlay configuration.
+* **generate_ci_job.py** generates a *CI* nightly job for for each platform
+  and architecture listed in the *CI build file*.
+* **generate_ci_script.py** generates a *shell* script which will run the
+  same tasks as the *CI* job on a local machine.
+
+The build process in detail
+---------------------------
+
+The actual build is performed within a Docker container in order to ensure
+only declared dependencies are available.
+
+The actual build process starts in the script *create_ci_task_generator.py*.
+It generates three Dockerfiles: one to perform the *create-workspace* task to
+populate the workspace and enumerate prerequisites, one to perform the
+*build-and-install* task, and one to perform the *build-and-test* task.
+
+A typical job runs 4 Dockerfiles in all.
+
+Create workspace
+^^^^^^^^^^^^^^^^
+
+This task is performed by the script *create_workspace.py*, the Colcon
+executable and the rosdep executable.
+
+The task performs the following steps:
+
+* Prepare the package sources
+
+  * Fetches the `.repos` file(s)
+  * Fetches the source repositories containing the package(s) to be built
+  * If necessary, switches the branch in those repositories
+
+* Select the packages which should be built, tested, and installed
+
+  * Marks the packages explicitly listed as ignored
+  * If necessary, ignores any packages not selected or in scope
```

Comment:
Yeah, this doesn't come across very clearly. My intent is to differentiate between the packages explicitly and unconditionally ignored, and the packages which are ignored because they aren't either selected (like `--packages-select`) or in scope (via `--depth-{before,after}`).

Reply:
I clarified this in light of the changes to the scoping mechanism.